packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -11,53 +19,102 @@ interface VersionData {
   latest: Version
 }
 
+const REMOTE_MANIFEST_URL = 'https://download.cypress.io/desktop.json'
+const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
+
+type GetLatestVersionOptions = { initialLaunch: boolean, testingType: TestingType | null, currentCypressVersion: string }
+
 export class VersionsDataSource {
-  static result: undefined | Promise<string>
+  constructor (private ctx: DataContext) {}
+
+  static npmMetadata: undefined | Record<string, string>
+  static initialLaunch: boolean = true
```

Comment:
Could be missing something, but any specific reason these are static members, as opposed to instance?

I wonder if `initialLaunch`, and possibly `npmMetadata`, should be in `coreData`. It seems like something that might be useful outside of this class. You'd then access it with `this.ctx.coreData.initialLaunch` and update with `this.ctx.update(coreData => coreData.initialLaunch = false)`. There's some general info here: https://github.com/cypress-io/cypress/tree/10.0-release/packages/data-context on how data-context works (still figuring out the patterns).

Reply:
Mainly these are static so that they can be tracked over the life of the application. Core data is an interesting thought. I’ll think about that.

Reply:
Changed with a refactor:

[`19b094d` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/19b094d70148ddc58e68b5c224f33e940137306a)

[`ae508f1` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/ae508f18262baeba7bd1a4c37b8dd93a9df804f5)